### PR TITLE
Update CSP for CodeEditor

### PR DESCRIPTION
**Pull request by zuoyuanh**
*Issue #, if available:*

*Description of changes:*

Update CSP for CodeEditor. The CSP is set based upon:
* Instructions in https://code.visualstudio.com/docs/setup/network
* Allowing https://open-vsx.org for extension marketplace
* Confirmed references to https://raw.githubusercontent.com in source code are server-side calls and should not be impacted by CSP update

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: patches/update-csp.diff [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
++++ sagemaker-code-editor/vscode/src/vs/server/node/webClientServer.ts
+@@ -375,7 +375,7 @@ export class WebClientServer {
+ 			`frame-src 'self' https://*.vscode-cdn.net data:;`,
+ 			'worker-src \'self\' data: blob:;',
+ 			'style-src \'self\' \'unsafe-inline\';',
+-			'connect-src \'self\' ws: wss: https:;',
++			'connect-src \'self\' ws: wss: https://main.vscode-cdn.net http://localhost:* https://localhost:* https://login.microsoftonline.com/ https://update.code.visualstudio.com https://*.vscode-unpkg.net/ https://default.exp-tas.com/vscode/ab https://vscode-sync.trafficmanager.net https://vscode-sync-insiders.trafficmanager.net https://*.gallerycdn.vsassets.io https://marketplace.visualstudio.com https://*.blob.core.windows.net https://az764295.vo.msecnd.net  https://code.visualstudio.com https://*.gallery.vsassets.io https://*.rel.tunnels.api.visualstudio.com wss://*.rel.tunnels.api.visualstudio.com https://*.servicebus.windows.net/ https://vscode.blob.core.windows.net https://vscode.search.windows.net https://vsmarketplacebadges.dev https://vscode.download.prss.microsoft.com https://download.visualstudio.microsoft.com https://*.vscode-unpkg.net https://open-vsx.org;',
```
Comment (sgganjo) [inline patches/update-csp.diff:10]: How did we figure out what domains are allowed

**Review by sgganjo — COMMENTED** [review]
How did we figure out what domains we are going to allow

**Comment by zuoyuanh** [comment]
> How did we figure out what domains we are going to allow

It is mainly based on instructions in https://code.visualstudio.com/docs/setup/network, along with allowing https://open-vsx.org/ for extension marketplace. Updated the description to include the info.

**Review by sgganjo — APPROVED** [review]

**Comment by sgganjo** [comment]
Do we need to add any steps around how to test this locally as well

**Review by aws-pangestu — APPROVED** [review]

**Comment by aws-pangestu** [comment]
> Do we need to add any steps around how to test this locally as well

Same here, could you at least add high level what you did to test this locally?

File: patches/update-csp.diff [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Index: sagemaker-code-editor/vscode/src/vs/server/node/webClientServer.ts
```
Comment (aws-prayags) [inline patches/update-csp.diff:1]: let's add a comment explaining why this patch is needed

**Comment by zuoyuanh** [comment]
Yes, here are the testings I performed:
1. Add `?payload=%5B%5B%22openFile%22%2C%22https://raw.githubusercontent.com/aws/studio-lab-examples/main/connect-to-aws/Access_AWS_from_Studio_Lab.ipynb%22%5D%5D`. Confirmed notebook is not loading and we are seeing error `The editor could not be opened due to an unexpected error`
2. Confirmed I am able to install extensions from marketplace
3. Performed basic testings for notebooks, terminals and files. Basic functionalities work.

**Pull request merged**